packages/codegen-ui-react/lib/forms/form-renderer-helper/bidirectional-relationship.ts

```diff
@@ -79,6 +79,116 @@ function getFieldBiDirectionalWith({
+              factory.createPropertyAccessExpression(
+                factory.createIdentifier(modelName),
+                factory.createIdentifier('copyOf'),
+              ),
+              undefined,
+              [
+                factory.createIdentifier(recordNameToUnlink),
+                factory.createArrowFunction(
+                  undefined,
+                  undefined,
+                  [
+                    factory.createParameterDeclaration(
+                      undefined,
+                      undefined,
+                      undefined,
+                      factory.createIdentifier('updated'),
+                      undefined,
+                      undefined,
+                      undefined,
+                    ),
+                  ],
+                  undefined,
+                  factory.createToken(SyntaxKind.EqualsGreaterThanToken),
+                  factory.createBlock(
+                    [
+                      factory.createExpressionStatement(
+                        factory.createBinaryExpression(
+                          factory.createPropertyAccessExpression(
+                            factory.createIdentifier('updated'),
+                            factory.createIdentifier(fieldName),
+                          ),
+                          factory.createToken(SyntaxKind.EqualsToken),
+                          factory.createIdentifier('undefined'),
+                        ),
+                      ),
+                      ...associatedFields.map((field) =>
+                        factory.createExpressionStatement(
+                          factory.createBinaryExpression(
+                            factory.createPropertyAccessExpression(
+                              factory.createIdentifier('updated'),
+                              factory.createIdentifier(field),
+                            ),
+                            factory.createToken(SyntaxKind.EqualsToken),
+                            factory.createIdentifier('undefined'),
+                          ),
+                        ),
+                      ),
+                    ],
+                    true,
+                  ),
+                ),
+              ],
+            ),
+          ],
+        ),
+      ],
+    ),
+  );
+}
+
+function unlinkModelThrowErrorExpression(
+  relatedModelName: string,
+  relatedRecordToLink: string,
+  modelName: string,
+  isUnlink?: boolean,
+) {
+  // "Image 398038 cannot be linked to FakeModel because it is already linked to another FakeModel"
+  return factory.createThrowStatement(
+    factory.createCallExpression(factory.createIdentifier('Error'), undefined, [
+      factory.createTemplateExpression(factory.createTemplateHead(`${relatedModelName} `), [
+        factory.createTemplateSpan(
+          factory.createPropertyAccessExpression(
+            factory.createIdentifier(relatedRecordToLink),
+            factory.createIdentifier('id'),
```

Comment:
Not all records have ids. If you look at the model of the relatedRecordToLink, it has an array of `primaryKeys`. You can just take the first one `primaryKeys[0]`